### PR TITLE
Lims 4264

### PR DESCRIPTION
**Pull request by mazenmmbadr**

### CODE REVIEW

**Comment by 0xIslamTaha** [comment]
```
xtremx@0xIslamTaha /opt/code/github/modeso/1lims-automation [LIMS-4264]
± % nosetests-3.4 -vs --logging-level=WARNING ui_testing/testcases/basic_tests/test03_orders.py:OrdersTestCases.test027_update_material_type --tc-file=config.ini --tc=site.password:admin                                                                                                                            !10523
Apply this on suborder number 5 for example:- ... 	
2019-07-30 18:01:23.443 | INFO     | ui_testing.testcases.base_test:setUp:25 - * Test case : test027_update_material_type
2019-07-30 18:01:24.868 | INFO     | api_testing.apis.base_api:info:52 - Get authorized api session.
2019-07-30 18:01:26.054 | INFO     | api_testing.apis.base_api:info:52 - session ID : eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7InVzZXJuYW1lIjoiYWRtaW4iLCJlbWFpbCI6ImFkbWluQG1vZGVzby5jaCIsInJvbGUiOiJBZG1pbiJ9LCJpYXQiOjE1NjQ1MDI0ODUsImV4cCI6MTU2NjIzMDQ4NSwiaXNzIjoiMWxpbXMifQ.i7NOb4OXdwS4uGKSJ0bEVAvJyurApWxG9GTYHW6-YZQ
2019-07-30 18:01:26.054 | INFO     | api_testing.apis.base_api:info:52 - Get authorized api session.
2019-07-30 18:01:27.452 | INFO     | api_testing.apis.base_api:info:52 - session ID : eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7InVzZXJuYW1lIjoiYWRtaW4iLCJlbWFpbCI6ImFkbWluQG1vZGVzby5jaCIsInJvbGUiOiJBZG1pbiJ9LCJpYXQiOjE1NjQ1MDI0ODcsImV4cCI6MTU2NjIzMDQ4NywiaXNzIjoiMWxpbXMifQ.lsNxvs8LeLX4Ol08U1z0HKtGaZ3Y71Lfj--jgog7Ccw
2019-07-30 18:01:27.453 | INFO     | api_testing.apis.base_api:info:52 - Get authorized api session.
2019-07-30 18:01:28.321 | INFO     | api_testing.apis.base_api:info:52 - session ID : eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7InVzZXJuYW1lIjoiYWRtaW4iLCJlbWFpbCI6ImFkbWluQG1vZGVzby5jaCIsInJvbGUiOiJBZG1pbiJ9LCJpYXQiOjE1NjQ1MDI0ODgsImV4cCI6MTU2NjIzMDQ4OCwiaXNzIjoiMWxpbXMifQ.ocIIJr7Y04CY0nl96lWLWn6EreK2x5UJ0t2eCxfV9dA
2019-07-30 18:01:28.322 | INFO     | api_testing.apis.base_api:info:52 - Get authorized api session.
2019-07-30 18:01:29.334 | INFO     | api_testing.apis.base_api:info:52 - session ID : eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjp7InVzZXJuYW1lIjoiYWRtaW4iLCJlbWFpbCI6ImFkbWluQG1vZGVzby5jaCIsInJvbGUiOiJBZG1pbiJ9LCJpYXQiOjE1NjQ1MDI0ODgsImV4cCI6MTU2NjIzMDQ4OCwiaXNzIjoiMWxpbXMifQ.ocIIJr7Y04CY0nl96lWLWn6EreK2x5UJ0t2eCxfV9dA
2019-07-30 18:01:29.335 | INFO     | ui_testing.pages.login_page:login:10 -  + Login admin : admin.
2019-07-30 18:02:39.351 | INFO     | ui_testing.pages.base_pages:sleep_small:36 -  + Small sleep.
2019-07-30 18:02:44.357 | INFO     | ui_testing.testcases.base_test:get_random_order_data:114 - Generate new random data to update the order with
2019-07-30 18:02:46.987 | INFO     | ui_testing.pages.base_pages:sleep_small:36 -  + Small sleep.
2019-07-30 18:02:51.993 | INFO     | ui_testing.pages.article_page:create_new_article:8 -  + Create new article.
2019-07-30 18:03:02.056 | INFO     | ui_testing.pages.base_pages:save:48 -  + Save the changes.
2019-07-30 18:03:12.160 | INFO     | ui_testing.pages.article_page:create_new_article:33 -  + Article name : e2b54f7568
2019-07-30 18:03:12.160 | INFO     | ui_testing.testcases.base_test:get_random_order_data:121 - Generate new article with article name: e2b54f7568, and material type: 34e2f5262e
2019-07-30 18:03:12.161 | INFO     | ui_testing.pages.testplans_page:get_test_plans_page:11 -  + Get test plans page.
2019-07-30 18:03:15.382 | INFO     | ui_testing.pages.base_pages:sleep_small:36 -  + Small sleep.
2019-07-30 18:03:20.386 | INFO     | ui_testing.pages.testplan_page:create_new_test_plan:54 -  + Create new test plan
2019-07-30 18:03:20.774 | INFO     | ui_testing.pages.base_pages:sleep_small:36 -  + Small sleep.
2019-07-30 18:03:28.523 | INFO     | ui_testing.pages.base_pages:sleep_tiny:32 -  + Tiny sleep.
2019-07-30 18:03:39.439 | INFO     | ui_testing.pages.base_selenium:select_item_from_drop_down:377 -  There is no Qualitative option in the drop-down
2019-07-30 18:03:39.565 | INFO     | ui_testing.pages.base_pages:save:48 -  + Save the changes.
2019-07-30 18:03:49.698 | INFO     | ui_testing.pages.testplan_page:create_new_test_plan:76 -  + Test plan name : 27fb24586a
2019-07-30 18:03:49.698 | INFO     | ui_testing.pages.testplans_page:get_testunits_in_testplans:28 - Filter by testplan name 27fb24586a
2019-07-30 18:04:49.814 | INFO     | ui_testing.testcases.base_test:tearDown:43 -  * TearDown. 	
ERROR

======================================================================
ERROR: Apply this on suborder number 5 for example:-
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/code/github/modeso/1lims-automation/ui_testing/testcases/basic_tests/test03_orders.py", line 1110, in test027_update_material_type
    basic_order_data = self.get_random_order_data()
  File "/opt/code/github/modeso/1lims-automation/ui_testing/testcases/base_test.py", line 125, in get_random_order_data
    testplan_testunits = self.testplans_page.get_testunits_in_testplans(test_plan_name=new_testplan_name)
  File "/opt/code/github/modeso/1lims-automation/ui_testing/pages/testplans_page.py", line 29, in get_testunits_in_testplans
    new_testplan_record = self.search(value=test_plan_name)
  File "/opt/code/github/modeso/1lims-automation/ui_testing/pages/base_pages.py", line 20, in search
    self.base_selenium.set_text(element='general:search', value=value)
  File "/opt/code/github/modeso/1lims-automation/ui_testing/pages/base_selenium.py", line 277, in set_text
    self.wait_until_element_located(element)
  File "/opt/code/github/modeso/1lims-automation/ui_testing/pages/base_selenium.py", line 153, in wait_until_element_located
    dom_element = self.find_element(element=element)
  File "/opt/code/github/modeso/1lims-automation/ui_testing/pages/base_selenium.py", line 101, in find_element
    element_value = self.driver.find_element(getattr(By, method), value)
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriver.py", line 978, in find_element
    'value': value})['value']
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python3.6/dist-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"id","selector":"generalSearch"}
  (Session info: chrome=73.0.3683.86)
  (Driver info: chromedriver=72.0.3626.69 (3c16f8a135abc0d4da2dff33804db79b849a7c38),platform=Linux 4.15.0-50-generic x86_64)


----------------------------------------------------------------------
Ran 1 test in 206.373s

FAILED (errors=1)

```

File: ui_testing/pages/order_page.py [hunk 1]
```diff
@@ -125,13 +125,7 @@ def create_existing_order(self, no='', material_type='', article='', contact='',
         
         for test_unit in test_units:
             self.set_test_unit(test_unit)    
-        if multiple_suborders > 0:
```
[the remark this answers is not in the document]
Comment (mazenmmbadr) [reply ui_testing/pages/order_page.py]: easier to create 5 suborders, since all the suborders testcases needs to be tested starting from the 3rd suborder+ so this way, it duplicates the first suborder according to the number of orders i need, to start my test case

File: ui_testing/testcases/base_test.py [hunk 2]
```diff
@@ -84,4 +108,30 @@ def get_active_tst_unit_with_material_type(self, search, material_type='Raw Mate
             if test_unit_dict['Type'] == search and material_type in test_unit_dict['Material Type']:
                 return test_unit_dict
         return {}
+    
+    # this function generates random order data and then redirectes the user to the order's page
+    def get_random_order_data(self):
```
[the remark this answers is not in the document]
Comment (mazenmmbadr) [reply ui_testing/testcases/base_test.py]: okey

[the remark this answers is not in the document]
Comment (mazenmmbadr) [reply ui_testing/testcases/base_test.py]: i added it here in the first place, because it wasn't fully related to order's function, as it generates the data but by creating records in other entities, so i needed a place close to helper file, so i added it in the base test (same as get active article function)
